File: xy11233/src/models/ImportRecord.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const IMPORT_TYPES = {
  REQUISITION_CSV: 'requisition_csv',
  INVENTORY_JSON: 'inventory_json',
  HAZARD_RULES: 'hazard_rules',
  REAGENTS: 'reagents',
  USERS: 'users'
};

const IMPORT_STATUSES = {
  PENDING: 'pending',
  PROCESSING: 'processing',
  COMPLETED: 'completed',
  PARTIAL_SUCCESS: 'partial_success',
  FAILED: 'failed'
};

const ImportRecord = sequelize.define('ImportRecord', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  import_no: {
    type: DataTypes.STRING(50),
    allowNull: false,
    unique: true,
    comment: '导入单号'
  },
  import_type: {
    type: DataTypes.ENUM(Object.values(IMPORT_TYPES)),
    allowNull: false,
    comment: '导入类型'
  },
  file_name: {
    type: DataTypes.STRING(255),
    allowNull: false,
    comment: '文件名'
  },
  file_path: {
    type: DataTypes.STRING(500),
    allowNull: true,
    comment: '文件路径'
  },
  total_records: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
    comment: '总记录数'
  },
  success_records: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
    comment: '成功记录数'
  },
  failed_records: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
    comment: '失败记录数'
  },
  status: {
    type: DataTypes.ENUM(Object.values(IMPORT_STATUSES)),
    allowNull: false,
    defaultValue: IMPORT_STATUSES.PENDING,
    comment: '导入状态'
  },
  operator_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    comment: '操作人ID'
  },
  operator_name: {
    type: DataTypes.STRING(100),
    allowNull: false,
    comment: '操作人姓名'
  },
  started_at: {
    type: DataTypes.DATE,
    allowNull: true,
    comment: '开始时间'
  },
  completed_at: {
    type: DataTypes.DATE,
    allowNull: true,
    comment: '完成时间'
  },
  error_message: {
    type: DataTypes.TEXT,
    allowNull: true,
    comment: '错误信息'
  },
  remark: {
    type: DataTypes.TEXT,
    allowNull: true,
    comment: '备注'
  }
}, {
  tableName: 'import_records',
  comment: '导入记录表'
});

ImportRecord.IMPORT_TYPES = IMPORT_TYPES;
ImportRecord.IMPORT_STATUSES = IMPORT_STATUSES;

module.exports = ImportRecord;

File: xy11262/app/models/import_record.py
import enum
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime, Enum, JSON, ForeignKey
from sqlalchemy.orm import relationship

from app.core.database import Base


class ImportStatus(str, enum.Enum):
    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"
    PARTIAL = "partial"


class ImportType(str, enum.Enum):
    HAZARD_CSV = "hazard_csv"
    PHOTO_JSON = "photo_json"
    REVIEW_RECORD = "review_record"


class ImportRecord(Base):
    __tablename__ = "import_records"

    id = Column(Integer, primary_key=True, index=True)
    import_type = Column(Enum(ImportType), nullable=False)
    status = Column(Enum(ImportStatus), default=ImportStatus.PENDING)
    
    file_name = Column(String(255), nullable=False)
    file_path = Column(String(500))
    file_size = Column(Integer)
    
    total_records = Column(Integer, default=0)
    success_count = Column(Integer, default=0)
    failed_count = Column(Integer, default=0)
    
    started_at = Column(DateTime)
    completed_at = Column(DateTime)
    
    imported_by = Column(String(100))
    remarks = Column(Text)
    
    created_at = Column(DateTime, default=datetime.utcnow)
    
    bad_records = relationship("BadRecord", back_populates="import_record", cascade="all, delete-orphan")


class BadRecord(Base):
    __tablename__ = "bad_records"

    id = Column(Integer, primary_key=True, index=True)
    import_record_id = Column(Integer, ForeignKey("import_records.id"), nullable=False)
    
    row_number = Column(Integer)
    original_data = Column(JSON, nullable=False)
    
    error_type = Column(String(100))
    error_message = Column(Text, nullable=False)
    
    suggested_fix = Column(Text)
    field_errors = Column(JSON)
    
    corrected = Column(Integer, default=0)
    correction_notes = Column(Text)
    
    created_at = Column(DateTime, default=datetime.utcnow)
    
    import_record = relationship("ImportRecord", back_populates="bad_records")
